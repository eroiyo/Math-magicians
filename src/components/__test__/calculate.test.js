/* eslint-disable linebreak-style */

import calculate from '../../logic/calculate';

let input = {};
describe('Calculator test', () => {
  it('testing the sum operation', () => {
    input = {
      total: 34, // this is the input format
      next: 18, // current save value
      operation: '+', // the operation
    };
    const result = calculate(input, '=');
    expect(result).toMatchSnapshot({
      total: '52',
      next: null,
      operation: null,
    });
  });

  it('test the result of +/-', () => {
    input = {
      total: 5,
      next: 9,
      operation: '-',
    };

    const result = calculate(input, '+/-');

    expect(result).toMatchSnapshot({
      next: '-9',
    });
  });
  it('testing the substration operation', () => {
    input = {
      total: 34,
      next: 18,
      operation: '-',
    };
    const result = calculate(input, '=');
    expect(result).toMatchSnapshot({
      total: '16',
      next: null,
      operation: null,
    });
  });
  it('testing the division operation', () => {
    input = {
      total: 4,
      next: 2,
      operation: 'รท',
    };
    const result = calculate(input, '=');
    expect(result).toMatchSnapshot({
      total: '2',
      next: null,
      operation: null,
    });
  });
  it('testing the multiplication operation', () => {
    input = {
      total: 4,
      next: 2,
      operation: 'x',
    };
    const result = calculate(input, '=');
    expect(result).toMatchSnapshot({
      total: '8',
      next: null,
      operation: null,
    });
  });
  it('testing the multiplication operation', () => {
    input = {
      total: 5,
      next: 5,
      operation: '%',
    };
    const result = calculate(input, '=');
    expect(result).toMatchSnapshot({
      total: '0',
      next: null,
      operation: null,
    });
  });
  it('testing the AC Button', () => {
    input = {
      total: 52356346235233,
      next: 55744574723465,
      operation: '+',
    };
    const result = calculate(input, 'AC');
    expect(result).toMatchSnapshot({
      total: '0',
      next: null,
      operation: null,
    });
  });
});
