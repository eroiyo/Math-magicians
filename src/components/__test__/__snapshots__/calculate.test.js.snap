// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator test test the result of +/- 1`] = `
Object {
  "next": "-9",
}
`;

exports[`Calculator test testing the AC Button 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "0",
}
`;

exports[`Calculator test testing the division operation 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "2",
}
`;

exports[`Calculator test testing the multiplication operation 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "8",
}
`;

exports[`Calculator test testing the multiplication operation 2`] = `
Object {
  "next": null,
  "operation": null,
  "total": "0",
}
`;

exports[`Calculator test testing the substration operation 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "16",
}
`;

exports[`Calculator test testing the sum operation 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "52",
}
`;
